[threaded-animation-resolution] don't accelerate transform interpolation on inline renderers

graouts opened the pull request:
#### e7ee3a4fe869d087dc98d27686c84c5e2343f2ee
<pre>
[threaded-animation-resolution] don&apos;t accelerate transform interpolation on inline renderers
<a href="https://bugs.webkit.org/show_bug.cgi?id=269568">https://bugs.webkit.org/show_bug.cgi?id=269568</a>

Reviewed by Dean Jackson.

Inline renderers don&apos;t support transform-related properties so there is no need to accelerate
interpolation of the transform property on such targets.

Since we now have several reasons why transform-related properties cannot be accelerated, we now
keep an `OptionSet&lt;AcceleratedEffectProperty&gt;` that indicates the set of accelerated properties that
were not allowed, passing it down to the `AcceleratedEffect` and removing the disallowed properties
all at once, including those that are marked at disallowed during construction of the `AcceleratedEffect`.

Now, when we encounter an inline renderer, we add the transform-related properties to the list
of disallowed properties.

* Source/WebCore/platform/animation/AcceleratedEffect.cpp:
(WebCore::AcceleratedEffect::create):
(WebCore::AcceleratedEffect::AcceleratedEffect):
(WebCore::AcceleratedEffect::validateFilters):
* Source/WebCore/platform/animation/AcceleratedEffect.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateAcceleratedEffectsAndBaseValues):

Canonical link: <a href="https://commits.webkit.org/274850@main">https://commits.webkit.org/274850@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/108733f851b993eaa8d61f1ce8ff437ccde0de09

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40204 "Failed to checkout and rebase branch from PR 24604") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19216 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/42582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/42749 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/36293 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/42511 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/22140 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16545 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/42749 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/40778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/22140 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/42582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/42749 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/22140 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/42582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/44027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/22140 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/42582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/44027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/15019 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/12315 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/44027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/16638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/42582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/16687 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5312 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/16282 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

grorg commented on Source/WebCore/rendering/RenderLayerBacking.cpp:
```diff
@@ -4075,12 +4081,10 @@ bool RenderLayerBacking::updateAcceleratedEffectsAndBaseValues()
-                if (blendingKeyframes.hasHeightDependentTransform() && animatesHeight)
-                    continue;
+                if ((blendingKeyframes.hasWidthDependentTransform() && animatesWidth) || (blendingKeyframes.hasHeightDependentTransform() && animatesHeight))
+                    disallowedAcceleratedProperties.add(transformRelatedAcceleratedProperties);
```
I liked it the original way.

But also, put the already calculated value before the call to the function so it can early return if necessary.

graouts replied:
Good point about using the already-computed `bool` flags first.

graouts committed:
[threaded-animation-resolution] don't accelerate transform interpolation on inline renderers
https://bugs.webkit.org/show_bug.cgi?id=269568

Reviewed by Dean Jackson.

Inline renderers don't support transform-related properties so there is no need to accelerate
interpolation of the transform property on such targets.

Since we now have several reasons why transform-related properties cannot be accelerated, we now
keep an `OptionSet<AcceleratedEffectProperty>` that indicates the set of accelerated properties that
were not allowed, passing it down to the `AcceleratedEffect` and removing the disallowed properties
all at once, including those that are marked at disallowed during construction of the `AcceleratedEffect`.

Now, when we encounter an inline renderer, we add the transform-related properties to the list
of disallowed properties.

* Source/WebCore/platform/animation/AcceleratedEffect.cpp:
(WebCore::AcceleratedEffect::create):
(WebCore::AcceleratedEffect::AcceleratedEffect):
(WebCore::AcceleratedEffect::validateFilters):
* Source/WebCore/platform/animation/AcceleratedEffect.h:
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateAcceleratedEffectsAndBaseValues):

Canonical link: https://commits.webkit.org/274850@main
